src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -111,48 +123,100 @@ static CallingConvention getCallingConvention(FunctionPointerCallingConventionSy
+                                if (!unmanagedList.ContainsDiagnostics)
+                                {
+                                    diagnostics.Add(ErrorCode.ERR_InvalidFunctionPointerCallingConvention, unmanagedList.OpenBracketToken.GetLocation(), "");
+                                }
+                                return CallingConvention.Default;
 
                             case { CallingConventions: var specifiers }:
-                                // PROTOTYPE(func-ptr): Handle multiple specifiers
-                                foreach (var convention in specifiers)
+                                // More than one identifier case
+                                checkUnmanagedSupport(typeBinder, callingConventionSyntax.ManagedOrUnmanagedKeyword.GetLocation(), diagnostics, isExtensionError: true);
+                                foreach (var specifier in specifiers)
                                 {
-                                    _ = reportBadConventionAndReturn(convention.Name.ValueText, convention.Name, diagnostics);
+                                    customModifiers.Add(handleIndividualUnrecognizedSpecifier(specifier, typeBinder, diagnostics));
                                 }
 
-                                return CallingConvention.Default;
+                                return CallingConvention.Unmanaged;
                         }
 
                     case var unexpected:
                         throw ExceptionUtilities.UnexpectedValue(unexpected);
                 }
 
-                static CallingConvention reportBadConventionAndReturn(string givenConvention, SyntaxNodeOrToken node, DiagnosticBag diagnostics)
+                static CallingConvention handleSingleConvention(FunctionPointerUnmanagedCallingConventionSyntax specifier, Binder typeBinder, ArrayBuilder<CustomModifier> customModifiers, DiagnosticBag diagnostics)
+                {
+                    checkUnmanagedSupport(typeBinder, specifier.GetLocation(), diagnostics, isExtensionError: true);
+                    customModifiers.Add(handleIndividualUnrecognizedSpecifier(specifier, typeBinder, diagnostics));
+                    return CallingConvention.Unmanaged;
+                }
+
+                static CustomModifier handleIndividualUnrecognizedSpecifier(FunctionPointerUnmanagedCallingConventionSyntax specifier, Binder typeBinder, DiagnosticBag diagnostics)
+                {
+                    const string CallingConventionMetadataFormatString = "System.Runtime.CompilerServices.CallConv{0}";
+                    string specifierText = specifier.Name.ValueText;
+                    var metadataName = string.Format(CultureInfo.InvariantCulture, CallingConventionMetadataFormatString, specifierText);
+
+                    var specifierType = typeBinder.Compilation.Assembly.CorLibrary.GetTypeByMetadataName(metadataName, includeReferences: false, isWellKnownType: true, conflicts: out _, warnings: null);
```

Comment:
>GetTypeByMetadataName [](start = 83, length = 21)

Do we have a test where there is a type with a matching name, but it is generic?

Reply:
Added `GenericCallingConventionType`

---
In reply to: [458925153](https://github.com/dotnet/roslyn/pull/46110#discussion_r458925153) [](ancestors = 458925153)